src/partners/thorchain.ts

```diff
@@ -0,0 +1,198 @@
+import { div } from 'biggystring'
+import { asArray, asNumber, asObject, asString, asUnknown } from 'cleaners'
+import fetch from 'node-fetch'
+
+import { PartnerPlugin, PluginParams, PluginResult, StandardTx } from '../types'
+import { datelog, smartIsoDateFromTimestamp } from '../util'
+
+const asThorchainTx = asObject({
+  date: asString,
+  in: asArray(
+    asObject({
+      address: asString,
+      coins: asArray(
+        asObject({
+          amount: asString,
+          asset: asString
+        })
+      ),
+      txID: asString
+    })
+  ),
+  out: asArray(
+    asObject({
+      address: asString,
+      coins: asArray(
+        asObject({
+          amount: asString,
+          asset: asString
+        })
+      ),
+      txID: asString
+    })
+  ),
+  pools: asArray(asString),
+  status: asString,
+  type: asString
+})
+
+const asApiKeys = asObject({
+  thorchainAddress: asString
+})
+
+const asSettings = asObject({
+  offset: asNumber
+})
+
+const asThorchainResult = asObject({
+  actions: asArray(asUnknown)
+})
+type Settings = ReturnType<typeof asSettings>
+type ThorchainResult = ReturnType<typeof asThorchainResult>
+
+const ROLLBACK = 500
+const LIMIT = 50
+const THORCHAIN_MULTIPLIER = 100000000
```

Comment:
RUNE_MULTIPLIER?

Reply:
It's entirely protocol wide on Thorchain, not specific to RUNE